chain/arweave/src/runtime/abi.rs

```diff
@@ -0,0 +1,170 @@
+        })
+    }
+}
+
+impl ToAscObj<AscTransactionArray> for Vec<codec::Transaction> {
+    fn to_asc_obj<H: AscHeap + ?Sized>(
+        &self,
+        heap: &mut H,
+        gas: &GasCounter,
+    ) -> Result<AscTransactionArray, DeterministicHostError> {
+        let content = self
+            .iter()
+            .map(|x| asc_new(heap, x, gas))
+            .collect::<Result<Vec<_>, _>>()?;
+        Ok(AscTransactionArray(Array::new(&*content, heap, gas)?))
+    }
+}
+
+impl ToAscObj<AscTagArray> for Vec<codec::Tag> {
+    fn to_asc_obj<H: AscHeap + ?Sized>(
+        &self,
+        heap: &mut H,
+        gas: &GasCounter,
+    ) -> Result<AscTagArray, DeterministicHostError> {
+        let content = self
+            .iter()
+            .map(|x| asc_new(heap, x, gas))
+            .collect::<Result<Vec<_>, _>>()?;
+        Ok(AscTagArray(Array::new(&*content, heap, gas)?))
+    }
+}
+
+impl ToAscObj<AscProofOfAccess> for codec::ProofOfAccess {
+    fn to_asc_obj<H: AscHeap + ?Sized>(
+        &self,
+        heap: &mut H,
+        gas: &GasCounter,
+    ) -> Result<AscProofOfAccess, DeterministicHostError> {
+        Ok(AscProofOfAccess {
+            option: asc_new(heap, &self.option, gas)?,
+            tx_path: asc_new(heap, self.tx_path.as_slice(), gas)?,
+            data_path: asc_new(heap, self.data_path.as_slice(), gas)?,
+            chunk: asc_new(heap, self.chunk.as_slice(), gas)?,
+        })
+    }
+}
+
+impl ToAscObj<AscTransaction> for codec::Transaction {
+    fn to_asc_obj<H: AscHeap + ?Sized>(
+        &self,
+        heap: &mut H,
+        gas: &GasCounter,
+    ) -> Result<AscTransaction, DeterministicHostError> {
+        Ok(AscTransaction {
+            format: self.format,
+            id: asc_new(heap, self.id.as_slice(), gas)?,
+            last_tx: asc_new(heap, self.last_tx.as_slice(), gas)?,
+            owner: asc_new(heap, self.owner.as_slice(), gas)?,
+            tags: asc_new(heap, &self.tags, gas)?,
+            target: asc_new(heap, self.target.as_slice(), gas)?,
+            quantity: asc_new(
+                heap,
+                self.quantity
+                    .as_ref()
+                    .map(|b| b.as_ref())
+                    .unwrap_or_default(),
+                gas,
+            )?,
+            data: asc_new(heap, self.data.as_slice(), gas)?,
+            data_size: asc_new(
+                heap,
+                self.data_size
+                    .as_ref()
+                    .map(|b| b.as_ref())
+                    .unwrap_or_default(),
+                gas,
+            )?,
+            data_root: asc_new(heap, self.data_root.as_slice(), gas)?,
+            signature: asc_new(heap, self.signature.as_slice(), gas)?,
+            reward: asc_new(
+                heap,
+                self.reward.as_ref().map(|b| b.as_ref()).unwrap_or_default(),
+                gas,
+            )?,
+        })
+    }
+}
+
+impl ToAscObj<AscBlock> for codec::Block {
```

Comment:
I think we should have the block without the actual transaction. 

The transaction should forwarded to the subgraph only via a `TransactionHandler` that would be invoked with the correct transaction the user asked for in the mapping.

Sending full blocks to the WASM is probably going to produce problem at scale, might work properly on low block num with a single subgraph, might explode with recent blocks when multiple subgraphs run.

Reply:
added `TransactionHandler`